### PR TITLE
[pallet-revive] tx fee fixes

### PR DESCRIPTION
**Pull request by pgherveou**
Apply some fixes to properly estimate ethereum tx fees:

- Set the `extension_weight` on the dispatch_info to properly calculate the fee with pallet_transaction_payment
- Expose the gas_price through Runtime API, just in case we decide to tweak the value in future updates, it should be read from the chain rather than be a shared constant exposed by the crate
- add a `evm_gas_to_fee` utility function to properly convert gas to substrate fee
- Fix some minor gas encoding for edge cases

### CODE REVIEW

File: substrate/frame/revive/src/evm/runtime.rs [hunk 1]
```diff
@@ -693,9 +692,8 @@ mod test {
 					log::debug!(target: LOG_TARGET, "Gas price: {:?}", tx.gas_price);
 				}))
 				.unwrap();
-
-		let expected_tip =
-			tx.gas_price.unwrap() * tx.gas.unwrap() - U256::from(GAS_PRICE) * tx.gas.unwrap();
-		assert_eq!(U256::from(extra.1.tip()), expected_tip);
+		let diff = tx.gas_price.unwrap() - U256::from(GAS_PRICE);
```
Comment (athei) [inline substrate/frame/revive/src/evm/runtime.rs:695]: Please no unchecked math in runtime. Let's be paranoid here.

Comment (pgherveou) [reply substrate/frame/revive/src/evm/runtime.rs:695]: that's a test though should be fine :upside_down_face:

**Review by athei — APPROVED** [review]

File: substrate/frame/revive/src/evm/runtime.rs [hunk 2]
```diff
@@ -55,7 +55,7 @@ type CallOf<T> = <T as frame_system::Config>::RuntimeCall;
 /// - Not too high, ensuring the gas value is large enough (at least 7 digits) to encode the
 ///   ref_time, proof_size, and deposit into the less significant (6 lower) digits of the gas value.
 /// - Not too low, enabling users to adjust the gas price to define a tip.
-pub const GAS_PRICE: u32 = 1_000u32;
+pub(crate) const GAS_PRICE: u64 = 1_000u64;
```
Comment (xermicus) [inline substrate/frame/revive/src/evm/runtime.rs:58]: This might be interesting to access for downstream consumers, why is it no longer public?

**Review by xermicus — APPROVED** [review]

File: substrate/frame/revive/src/evm/runtime.rs (hunk 2 above)
Comment (xermicus) [reply substrate/frame/revive/src/evm/runtime.rs:58]: Nevermind, it doesn't matter because the getter is public.
